src/dataMigration.js

```diff
@@ -73,6 +73,27 @@ const dataMigrations = [
           }
         });
       });
+
+      const nameResults = database.objects('Name').filtered('id = $0',
+                                settings.get(SETTINGS_KEYS.SUPPLYING_STORE_NAME_ID));
+      if (nameResults.length < 1) throw new Error('Supplying Store Name ID missing from settings');
+      const mainSupplyingStoreName = nameResults[0];
+
+      const requisitions = database.objects('Requisition')
+                                   .filtered('otherStoreName = Null').snapshot();
+
+      database.write(() => {
+        requisitions.forEach(requisition => {
+          if (!requisition.isRequest) return;
+          // The reason for database.update is we don't want to update finalized transactions
+          // and incase database.save is replaced by automatic way of invoking db event listeners
+          // simply assigning property may cause changes to finalized transaction to be synced.
```

Comment:
Should not re sync finalised requisitions as it will cause v2, v3 to replicate requisition. And as for otherStoreName (i.e name_ID), mSupply has migration code that already assigns correct name_ID.